Payments over the Phone Guide (#3767)
Nexmo/nexmo-developer: _documentation/en/voice/voice-api/guides/payments.md

---
title: Payments over the Phone
description: Take payments during phone calls in a PCI compliant secure way
navigation_weight: 11
---

# Payments over the Phone [Developer Preview]

There are various scenarios when you may want to charge the user during the phone call, for example, automated order processing, subscription renewal, debt collection. To do that securely with credit cards, the application which deals with sensitive card data (numbers, expiration dates, security codes) must follow PCI compliance rules. Vonage Voice API provides you with the option to seamlessly embed secure payment processing into the call flow using the [NCCO](/voice/voice-api/guides/ncco) syntax.

> Payments over the Phone are currently offered as [Developer Preview](/product-lifecycle/dev-preview). The feature can be used in your projects for supported scenarios (see below). The following limitations apply:
<ul style='list-style:disc;margin-left:16px;margin-top:16px;'><li style='margin-bottom:16px;'>payment gateway configuration is being done by request,</li>

> <li style='margin-bottom:16px;'>`pay` action supported for inbound calls only,</li>

Reviewer: ```suggestion
<li style='margin-bottom:16px;'>The `pay` action supported for inbound calls only</li>
```

> <li style='margin-bottom:16px;'>US dollars supported as the only currency,</li>

Reviewer: ```suggestion
<li style='margin-bottom:16px;'>US dollars are the only supported currency</li>
```

> <li style='margin-bottom:16px;'>Visa and Mastercard cards supported only,</li>

Reviewer: ```suggestion
<li style='margin-bottom:16px;'>Only Mastercard and Visa cards supported</li>
```

> <li style='margin-bottom:16px;'>[Stripe](https://www.stripe.com) supported as the only payment gateway,</li>

Reviewer: ```suggestion
<li style='margin-bottom:16px;'>[Stripe](https://www.stripe.com) is the only supported payment gateway</li>
```

> <li style='margin-bottom:16px;'>the feature is supported free of extra charge during Developer Preview.</li></ul>

Reviewer: ```suggestion
<li style='margin-bottom:16px;'>The feature is supported free of extra charge during Developer Preview.</li></ul>
```

## Preliminary Configuration

> To start, you should have your Stripe account ready. If you don't have an account yet, [sign up here](https://dashboard.stripe.com/register).

Reviewer: Is this whole section done via emailing? or someone emails then they can make the request to turn on payments?

> > In [Developer Preview](/product-lifecycle/dev-preview), Vonage will set up your payment gateway connector for you. [Contact us](mailto:pay.voice@vonage.com) to get it configured.

Reviewer: ```suggestion
> In [Developer Preview](/product-lifecycle/dev-preview), Vonage will set up your payment gateway connector for you. [Contact us](mailto:pay.voice@vonage.com) to get it configured. Once this is done, continue with the following steps.
```

                "event_url": {
                    "address": "https://example.com",
                    "http_method": "POST"
                },
                "fallback_answer_url": {
                    "address": "",
                    "http_method": "GET"
                },
                "answer_url": {
                    "address": "https://example.com",
                    "http_method": "GET"
                }
            }
        }
    },
    "_links": {
        "self": {
            "href": "/v2/applications/YOUR_APPLICATION_ID"
        }
    }
}
```

> Unlike Voice API, the Applications API uses [header-based API Key and Secret Authentication] (https://developer.nexmo.com/concepts/guides/authentication#header-based-api-key-and-secret-authentication), which means you should use a [Base64](https://tools.ietf.org/html/rfc4648#section-4) encoded API key and secret joined by a colon in the `Authorization` header of the HTTP request.

### Update Application

Update your application with a [Update an application](https://developer.nexmo.com/api/application.v2#updateApplication) HTTP request:

```http
PUT https://api.nexmo.com/v2/applications/YOUR_APPLICATION_ID
```

> Use the response JSON from the previous step as the request body with the addition of the `signed_callbacks` parameter:

Reviewer: ```suggestion
Use the response JSON from the previous step as the request body with the addition of the `payment_enabled ` parameter:
```

                    "http_method": "GET"
                }
            },
            "payment_enabled": true
        }
    },
    "_links": {
        "self": {
            "href": "/v2/applications/YOUR_APPLICATION_ID"
        }
    }
}
```

You can make a `GET` request from step one again to ensure the parameter is applied (it should be returned in the response).

> Developer Preview limitation: if you change any parameter of your application via [Dashboard](https://dashboard.nexmo.com), the `payment_enabled` parameter will be dropped, and the feature will be inactivated, so you have to go through the activation steps again to turn it back on.

## Pay Action

To start the secure payment IVR, use the `pay` [NCCO action](/voice/voice-api/guides/ncco):

```json
[
  {
     "action": "pay",
     "amount": 9.99
  }
]
```

This will trigger the IVR flow with the following prompts:
- *Please enter your long card number*
- *Please enter your four-digit credit card expiration date*
- *Please enter your three-digit credit card security code*

In case the user enters the card data correctly, they will be charged $9.99, and your application will get the following callback:

> ```json

Reviewer: is it possible for this response to have the stripe ID or something>
